### PR TITLE
Refactor testDatePickerScripts in instructorFeedbackPageUiTest.java #3759

### LINKED ISSUE #3759: Refactor testDatePickerScripts in instructorFeedbackPageUiTest.java
Introduce explaining variables for the various asserts in that test
For example, in

<code>
assertEquals(sdf.format(cal.getTime()),
                     feedbackPage.getValueOfDate(Const.ParamsNames.FEEDBACK_SESSION_VISIBLEDATE));
        assertEquals(sdf.format(cal.getTime()),
                     feedbackPage.getMaxDateOf(Const.ParamsNames.FEEDBACK_SESSION_VISIBLEDATE));
        assertEquals(sdf.format(cal.getTime()),
                     feedbackPage.getMinDateOf(Const.ParamsNames.FEEDBACK_SESSION_PUBLISHDATE));
</code>

The code are stacked together and look extremely similar, making the subject of the tests unclear.
Variables can be introduced to explain what the check is checking for. This will help to increase the readability of the assertions.

In addition, it might help to make the dates in the different test segments independent of each other by reinitialising the dates or creating new date variables at the start of every segment, since a significant amount of date manipulation occur between each test segment. This makes it hard to trace and keep track of the values of the date between the different test segments.

### PR DESCRIPTION
**Pull request by nsaiisasidhar**
Fixes #3759

Please review the PR and let me know if any more changes are needed.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@nsaiisasidhar `cal_TS[1-6]` is not an acceptable variable name so you need to fix them.

**Comment by nsaiisasidhar** [comment]
@wkurniawan07 Sure. I'll check the documentation and change the variable names accordingly. Please let me know whether all the other changes are good.

**Comment by wkurniawan07** [comment]
@nsaiisasidhar it is on the README page. Quoting the relevant portion: "Variable names must be in mixed case starting with lower case.". In regex form this would be `[a-z][a-zA-Z0-9]*`, and is enforced via CheckStyle in our build tool (thus you observed the red "X" marking an errored build).

That aside, there are two problems that may occur:
1. `cal_TS[1-6]` does not immediately make it clear what each calendar variable is supposed to be about (which is what the issue is about). Something like `baseCal`, `calBefore`, `calAfter` (these are just examples) would be clearer in expressing that one calendar is used as base, and the other modified based on it. Another alternative would be `cal[1-6]` where they are chronologically arranged (make sure this is documented).
2. `Calendar cal_TS1 = cal` is assigning by reference, not value. This means if `cal_TS1` is modified, `cal` will also be modified.

I will review more thoroughly once Travis gives a passing build (a green tick).

**Comment by nsaiisasidhar** [comment]
@wkurniawan07 Can you please let me know where I can check error log file to know what defects in code caused travis build failure.

**Comment by wkurniawan07** [comment]
@nsaiisasidhar any updates here or any difficulties faced?

**Comment by nsaiisasidhar** [comment]
@wkurniawan07 I tried reinitializing the date variable at start to each test segment. But, there is an assertion test failing at line 751, when the **cal** variable is reinitialized to date **16**, which is otherwise changed to **11** by the two test segments before it.

**Comment by wkurniawan07** [comment]
Hi @nsaiisasidhar, any updates here?

**Comment by nsaiisasidhar** [comment]
@wkurniawan07 Is it fine, if the date variable is initialized with hard coded values in each test segment to make it independent.

**Comment by wkurniawan07** [comment]
> Is it fine, if the date variable is initialized with hard coded values in each test segment to make it independent.

@nsaiisasidhar indeed that is what should have been done.

**Comment by wkurniawan07** [comment]
@nsaiisasidhar please make sure that there is no compile error before pushing to GitHub. Eclipse should be able to tell if there are compile errors.

**Comment by nsaiisasidhar** [comment]
@wkurniawan07 In the last commit, I made  everything looks fine, indeed it is for initializing the _valueOfStartDate_ variable. But, why is that still showing compile error at line 755,(that variable is not initialized) when it is being actually initialized in line 755.

**Comment by nsaiisasidhar** [comment]
@wkurniawan07 Hi, can you please review it now.

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java [hunk 1]
```diff
@@ -694,12 +694,20 @@ public void testDatePickerScripts() {
         feedbackPage.clickCustomVisibleTimeButton();
         feedbackPage.clickCustomPublishTimeButton();
         
+        // Variables
+        String valueOfStartDate;
+        String valueOfEndDate;
+        String valueOfVisibleDate;
+        String maxValueOfVisibleDate;
+        String minValueOfPublishDate;
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java]: You don't have to define these variables beforehand. Defining them just prior to their first usage is fine.

**Comment by wkurniawan07** [comment]
@nsaiisasidhar there are merge conflicts that need to be resolved here.

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java [hunk 2]
```diff
@@ -694,12 +694,14 @@ public void testDatePickerScripts() {
         feedbackPage.clickCustomVisibleTimeButton();
         feedbackPage.clickCustomPublishTimeButton();
         
+
```
Comment (samsontmr) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java]: Try not to add whitespaces unnecessarily

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java [hunk 3]
```diff
@@ -694,12 +694,14 @@ public void testDatePickerScripts() {
...
         feedbackPage.clickCustomPublishTimeButton();
         
+        
         // setup various dates
         Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
         SimpleDateFormat sdf = new SimpleDateFormat("EEE MMM dd yyyy");
         sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
         cal.set(2014, 3, 16, 0, 0, 0);
         
+
```
Comment (samsontmr) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java]: Same here

**Comment by nsaiisasidhar** [comment]
I fixed logic errors, but still see that its conflicting. Can someone please suggest how to resolve the conflicts.

**Comment by chowyb** [comment]
@nsaiisasidhar you need to merge the `master` branch of the committer repo (which should be the `upstream` remote if you set it up correctly) into your branch and fix any resulting merge conflicts.

**Comment by nsaiisasidhar** [comment]
@chowyb Hi, I merged the master at commit # 8302a59.

**Comment by wkurniawan07** [comment]
@nsaiisasidhar merging with `master` in the past does not guarantee that conflicts will not happen again in the future. You have to merge with the latest `master` again.

**Comment by nsaiisasidhar** [comment]
Closing the pull request to resolve differences between the two branches.

**Pull request closed without merging**